test/suites/depend_different_headers.bash

```diff
@@ -0,0 +1,178 @@
+SUITE_depend_different_headers_SETUP() {
+    unset CCACHE_NODIRECT
+
+    BASEDIR=`pwd`
+    BASEDIR1="${BASEDIR}/test/dir1/"
+    BASEDIR2="${BASEDIR}/test/dir2/"
+    BASEDIR3="${BASEDIR}/test/dir3/"
+    BASEDIR4="${BASEDIR}/test/dir4/"
+
+    mkdir -p ${BASEDIR1} ${BASEDIR2} ${BASEDIR3} ${BASEDIR4}
+
+    cat <<EOF >${BASEDIR1}/test.c
+#include "header.h"
+#include <stdio.h>
+
+void test(){
+#ifdef CHANGE_THAT_AFFECTS_OBJECT_FILE
+printf("with change");
+#else
+printf("no change");
+#endif
+}
+EOF
+    cp -f "${BASEDIR1}/test.c" ${BASEDIR2}
+    cp -f "${BASEDIR1}/test.c" ${BASEDIR3}
+    cp -f "${BASEDIR1}/test.c" ${BASEDIR4}
+
+    cat <<EOF >"${BASEDIR1}/header.h"
+void test();
+EOF
+
+    cat <<EOF >"${BASEDIR2}/header.h"
+#define CHANGE_THAT_AFFECTS_OBJECT_FILE
+void test();
+EOF
+
+    cat <<EOF >"${BASEDIR3}/header.h"
+#define CHANGE_THAT_DOES_NOT_AFFECT_OBJECT_FILE
+void test();
+EOF
+
+    cat <<EOF >"${BASEDIR4}/header.h"
+#include "header2.h"
+void test();
+EOF
+    cat <<EOF >"${BASEDIR4}/header2.h"
+static void some_function(){};
+EOF
+
+    backdate "${BASEDIR1}/header.h" "${BASEDIR1}/test.c"  "${BASEDIR2}/header.h" "${BASEDIR2}/test.c" "${BASEDIR3}/header.h" "${BASEDIR3}/test.c" "${BASEDIR4}/header.h" "${BASEDIR4}/header2.h" "${BASEDIR4}/test.c"
+
+    DEPFLAGS="-MD -MF test.d"
+}
+
+function generate_reference_compiler_output() {
```

Comment:
[Please remove the function keyword](http://mywiki.wooledge.org/BashPitfalls#function_foo.28.29).